src/sinks/influxdb/mod.rs

```diff
@@ -442,12 +446,12 @@ pub mod test_util {
     // 1542182950000000011
     //
     pub(crate) fn split_line_protocol(line_protocol: &str) -> (&str, &str, String, &str) {
-        let mut split = line_protocol.splitn(2, ',').collect::<Vec<&str>>();
-        let measurement = split[0];
-
-        split = split[1].splitn(3, ' ').collect::<Vec<&str>>();
+        // tags and ts may not be present
+        let parts: Vec<&str> = line_protocol.splitn(2, ' ').collect();
+        let (measurement, tags) = parts[0].split_once(',').unwrap_or((parts[0], ""));
+        let (fields, ts) = parts[1].split_once(' ').unwrap_or((parts[1], ""));
```

Comment:
Would be better to use `split_once` instead of `splitn(2)` here, as it doesn't need to allocate a new `Vec` for the results. However, as a test it doesn't really matter.

Reply:
Thanks @bruceg , they are all nice suggestions, and I'm learning from it. I'll update them soon.